### PR TITLE
124/fix-webapp-dependencies

### PR DESCRIPTION
**Pull request by YanisHerne**
# Cleaned npm dependencies & switched react-scripts to vite
* Uninstalled unused testing libraries
* Uninstalled typescript and related dependencies
* Migrated from outdated 'react-scripts' package to 'vite' for faster bundling, better maintainability, and no npm dependency vulnerabilities
* Removed unused templated files like 'tsconfig.json' and 'public/robots.txt'
* Everything works as before by running '. startup.sh'
* Generally improved the two README.md files

### CODE REVIEW

**Comment by YanisHerne** [comment]
Don't pull yet there's a one character typo in startup.sh that I need to fix.

File: web_app/app.py [hunk 1]
```diff
@@ -32,6 +33,9 @@ class PIDConstants(BaseModel):
 
 app = FastAPI()
 
+# Serve the static react frontend
+app.mount("/static", StaticFiles(directory="public"), name="public")
```
Comment (miyatakazuya) [inline web_app/app.py:37]: Is the directory intended to be `auv_gui/public`?

Running startup.sh yields this error:
`Running app.py Server
Traceback (most recent call last):
  File "/home/kamiyata/yonderdeep/Nautilus/web_app/app.py", line 37, in <module>
    app.mount("/static", StaticFiles(directory="public"), name="public")
  File "/home/kamiyata/.local/lib/python3.10/site-packages/starlette/staticfiles.py", line 56, in __init__
    raise RuntimeError(f"Directory '{directory}' does not exist")
RuntimeError: Directory 'public' does not exist
`

File: web_app/startup.sh [hunk 2]
```diff
@@ -1,16 +1,12 @@
-
 # !/bin/bash
 
 # Run Backend 
-echo "Running app.py..."
-python3 app.py &
```
Comment (miyatakazuya) [inline web_app/startup.sh:6]: Is there a reason to removing the control operator `&`? 

If you fix app.py, without the operator the frontend doesn't run.

**Review by miyatakazuya — CHANGES_REQUESTED** [review]
I checked out to confirm the changes but ran into a few errors trying to get startup.sh to run without errors. 
- Check my comments on app.py and startup.sh

File: web_app/startup.sh (hunk 2 above)
Comment (YanisHerne) [reply web_app/startup.sh:6]: Yeah that was the typo I was talking about

File: web_app/app.py (hunk 1 above)
Comment (YanisHerne) [reply web_app/app.py:37]: Yes it is intended to be that, will fix.

**Comment by YanisHerne** [comment]
Ok the app.py change is actually meant to be for the branch and PR for the other issue, so I only changed the .gitignore and the startup.sh

**Comment by YanisHerne** [comment]
Ok, should be good now

**Review by abijitj — APPROVED** [review]
Looks good to me. There was a merge conflict with the README file that I had to fix. But love to see the 0 vulnerabilities on npm install.

**Pull request merged**